codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/ServerHttpProtocolGenerator.kt

```diff
@@ -869,7 +870,7 @@ private class ServerHttpProtocolImplGenerator(
 
     // TODO These functions can be replaced with the ones in https://docs.rs/aws-smithy-types/latest/aws_smithy_types/primitive/trait.Parse.html
     private fun generateParseStrAsPrimitiveFn(binding: HttpBindingDescriptor): RuntimeType {
-        val output = symbolProvider.toSymbol(binding.member)
+        val output = symbolProvider.toSymbol(binding.member).makeOptional()
```

Comment:
I had to fix that, but I hope this is also ok for client codegen too

Reply:
I am not sure about the implications of this, I will have to check with the folks working on client codegen.

Reply:
For context this change was needed to set the value in operation_deser for primitive types like bool/i32/etc 
For example this generated code:
```
                input = input.set_fetch_owner(
                    crate::operation_deser::parse_str_list_objects_v2_input_fetch_owner(v)?,
                );
```
did not compile as the setter expected an Option<bool>.
I think this is actually a must have change anyhow because the generated body always returns Ok(Some(value)) and just the return type of the function was not matching.
See here - https://github.com/awslabs/smithy-rs/blob/6325c8a277225e786920d42df935371d27b11fc4/codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/ServerHttpProtocolGenerator.kt#L883

Reply:
@rcoh can you help us here?

Reply:
I think we can go with makeOptional() as this stage. Input validation is on the road map and it is something we will address later.